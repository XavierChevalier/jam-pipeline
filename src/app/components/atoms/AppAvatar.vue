<script lang="ts">
  import { defineComponent } from 'vue'
  import { SizeMixin } from '@/app/mixins/SizeMixin'

  export default defineComponent({
    mixins: [
      SizeMixin({
        xs: 'w-4.5 h-4.5', // TODO
        sm: 'w-8 h-8',
        base: 'w-10 h-10', // TODO
        lg: 'w-7 h-7', // TODO
      }),
    ],
  })
</script>

<script setup lang="ts">
  import { computed } from 'vue'
  import { string } from 'yup'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'

  const props = defineProps({
    userName: {
      type: String,
      required: true,
    },
    url: {
      type: String,
      required: true,
      validator: isPropertyValid(string().required().url()),
    },
  })

  const avatarAlt = computed(() => `The ${props.userName}'s avatar`)
</script>

<template>
  <img
    class="box-content p-0.5 rounded-full border-2 border-neutral-light"
    :class="sizeClass"
    :alt="avatarAlt"
    :src="url"
  />
</template>
