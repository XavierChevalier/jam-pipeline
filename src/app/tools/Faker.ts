export const generateImageUrl = (size = 50) => `https://picsum.photos/${size}`
