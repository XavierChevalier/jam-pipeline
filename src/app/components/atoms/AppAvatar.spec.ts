import * as Stories from './AppAvatar.stories'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'
import { generateImageUrl } from '@/app/tools/Faker'

describe('AppAvatar', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories)
    .snapshotEachStories()
    .property<string>('url', (property) => {
      property
        .itShouldBeDefined()
        .itShouldHaveAValidatorFunction()
        .itShouldSuccessWith(generateImageUrl())
        .itShouldFailWith('empty string', '')
        .itShouldFailWith('null', null)
    })
    .property<string>('size', (property) => {
      property
        .itShouldBeDefined()
        .itShouldHaveADefaultValue()
        .itShouldHaveAValidatorFunction()
        .itShouldSuccessWith('sm')
        .itShouldSuccessWith('base')
        .itShouldSuccessWith('lg')
        .itShouldFailWith('unknown size type', 'unknown')
    })
})
