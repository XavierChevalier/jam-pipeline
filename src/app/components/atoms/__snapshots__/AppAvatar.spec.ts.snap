// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAvatar stories snapshots [snapshot] should render the "PicsumPhoto" component story correctly 1`] = `
<img
  alt="The TheBaronless's avatar"
  class="box-content p-0.5 rounded-full border-2 border-neutral-light w-10 h-10"
  src="https://picsum.photos/50"
/>
`;
